history/ephemeral.go

```diff
@@ -0,0 +1,88 @@
+package history
+
+import (
+	"bytes"
+	"fmt"
+
+	gcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/zen-eth/shisui/beacon"
+)
+
+func (h *Network) isEphemeralOfferType(contentKey []byte) bool {
+	return ContentType(contentKey[0]) == OfferEphemeralType
+}
+
+func (h *Network) handleEphemeralContents(contentKeys [][]byte, contents [][]byte) error {
+	var parentHash gcommon.Hash
+	gotHead := false
+	headHash, err := h.externalOracle.GetHeadHash()
+	if err != nil {
+		return err
+	}
+	for i, content := range contents {
+		contentKey := contentKeys[i]
+		if !h.isEphemeralOfferType(contentKey) {
+			return fmt.Errorf("content key different of type Ephemeral: content key %x", contentKey)
+		}
+
+		header, err := DecodeBlockHeader(content)
```

Comment:
I feel this is not correct, the content should be a new dedicated content type for Ephemeral findcontent